src/camera.py

```diff
@@ -3,209 +3,182 @@
-        :type degree: float
+        :param degree: The number of degrees.
+        :type degree: int
         """
-        self._tilt_offset = self.validate_degree(degree, self.MIN_CALI_DEGREE,
-                                                         self.MAX_CALI_DEGREE)
-        self._DB.set('tilt_offset', self._tilt_offset)
+        self._tilt_offset = int(numpy.clip(degree, self.MIN_CALI_DEGREE,
+                                                   self.MAX_TILT_CALI_DEGREE))
+        self._db.set('tilt_offset', self._tilt_offset)
 
 
-    def get_tilt(self):
+    def get_tilt_angle(self):
         """
+        Gets the current tilt angle of the camera.
+
         :return: The current tilt angle (in degrees).
         :rtype: float
         """
         return self._current_tilt_angle
 
 
-    def validate_degree(self, degree, min_degree, max_degree):
-        """
-        Takes input and restricts to specifcied min and max.
-
-        Returns the value in the degree parameter that is bounded by the 
-        min_degree and max_degree parameters.
-
-        :param degree: The value to be bounded between min and max degree.
-        :type degree: float
-
-        :param min_degree: The lower bound of the validation.
-        :type min_degree: float
-
-        :param max_degree: the upper bound of the validation.
-        :type max_degree: float
-
-        :return: The value of degree bounded between the min and max parameters.
-        :rtype: float
-        """
-        return max(min(max_degree, degree), min_degree)
-
-
-    def test_pan(self):
-        """
-        Tests the panning of the camera
-
-        sets the camera to 90 degrees then to right most value at 0 degrees. 
-        Then pans the camera from right to left giving over the maximum
-        values showing validation function working as intended, ending at 180 
-        degrees. Then pans the camera from left to right ending at 0 degrees.
-        resets the camera to 90 degrees.
-        """
-        self.set_pan(90)
-        time.sleep(.2)
-        self.set_pan(0)
-        time.sleep(.2)
-        print ("panning from right to left")
-        for i in range (-20, 200, 2):
-            self.set_pan(i)
-            time.sleep(.04)
-
-        print ("panning from left to right")
-        for i in range (200, -20, -2):
-            self.set_pan(i)
-            time.sleep(.04)
-        
-        time.sleep(.2)
-        self.set_pan(90)
-
-
 def main():
+    """
+    Demonstrates the camera's panning and tilting functionality.
+
+    Creates object and pans from 0 to 180, then from 180 to 0. Then tilts from
+    0 to 150, then 150 to 0.
+    """
+    long_interval = .2
+    short_interval = .04
     cam = Camera()
-    cam.test_pan()
+
+    cam.pan(90)
+    time.sleep(long_interval)
+    cam.pan(0)
+    time.sleep(long_interval)
+    # Tests camera pan.
+    print "Panning from right to left."
+    for i in range (cam.MIN_PAN_DEGREE, cam.MAX_PAN_DEGREE, 2):
```

Comment:
No space between range and the parameters. Range is a function and that's weird to do. Also range isn't inclusive on the end condition (i.e. [start, stop) ). It doesn't make a huge difference here but for completeness and as an example, we should fix it.

Reply:
Everywhere else you use range too.